WordPress/Sniffs/XSS/EscapeOutputSniff.php

```diff
@@ -111,14 +111,24 @@ class WordPress_Sniffs_XSS_EscapeOutputSniff extends WordPress_Sniff {
 	 * @return array
 	 */
 	public function register() {
-		return array(
+		$tokens =  array(
 			T_ECHO,
 			T_PRINT,
 			T_EXIT,
 			T_STRING,
-			T_OPEN_TAG_WITH_ECHO,
 		);
 
+		// We can successfully report on unescaped open_short_tags only if they are enabled.
+		if ( true === (bool) ini_get( 'short_open_tag' ) ) {
```

Comment:
@david-binda This is the issue for PHP 5.4+ as the `T_OPEN_TAG_WITH_ECHO` is supported independently of the `short_open_tag` setting from PHP 5.4+ upwards, so basically, leave the token in the base array and only use the `else` clause of this part with the reversed condition and you should be fine.